src/coreComponents/codingUtilities/Section.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Section.hpp
+ */
+
+#ifndef GEOS_COMMON_SECTION_HPP
+#define GEOS_COMMON_SECTION_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class Section
+{
+public:
+
+  Section();
+
+  /**
+   * @brief Set the name of the section
+   * @param m_sectionTitle The name of the section
+   */
+  void setName( string_view m_sectionTitle );
```

Comment:
For me, it feels strange that the name of a section is not `const` and not given at construction:
- what is the purpose of an instance?
- what happens if the name is reset between begin() and end()?

Reply:
move to constructor